helpers/browser/driver.js

```diff
@@ -39,183 +42,105 @@ class ChromeProtocol {
       'disabled-by-default-v8.cpu_profile'
     ];
 
-    this._traceEvents = [];
-    this._currentURL = null;
-    this._instance = null;
-    this._networkRecords = [];
-    this._networkRecorder = null;
-
-    this.getPageHTML = this.getPageHTML.bind(this);
-    this.evaluateFunction = this.evaluateFunction.bind(this);
-    this.evaluateScript = this.evaluateScript.bind(this);
-    this.getServiceWorkerRegistrations =
-        this.getServiceWorkerRegistrations.bind(this);
-    this.beginTrace = this.beginTrace.bind(this);
-    this.endTrace = this.endTrace.bind(this);
-    this.beginNetworkCollect = this.beginNetworkCollect.bind(this);
-    this.endNetworkCollect = this.endNetworkCollect.bind(this);
+    this.timeoutID = null;
   }
 
-  get WAIT_FOR_LOAD() {
-    return true;
+  get url() {
+    return this._url;
   }
 
-  getInstance() {
-    if (!this._instance) {
-      this._instance = new Promise((resolve, reject) => {
-        // @see: github.com/cyrus-and/chrome-remote-interface#moduleoptions-callback
-        const OPTIONS = {};
-        chromeRemoteInterface(OPTIONS,
-          resolve
-        ).on('error', e => reject(e));
-      });
-    }
+  set url(_url) {
+    this._url = _url;
+  }
 
-    return this._instance;
+  connect() {
+    return new Promise((resolve, reject) => {
+      if (this._chrome) {
```

Comment:
this brings back the old race condition if multiple callers call `connect` before `this._chrome` is resolved, which is easy to do if two methods call it without you knowing.